src/app/auth/auth.component.ts

```diff
@@ -296,4 +297,15 @@ export class AuthComponent implements OnInit, OnDestroy {
       });
     });
   }
+
+  private createProfileFromUrlQueryParams() {
+    if (window.location.search) {
+      const urlParams = new  URLSearchParams(window.location.search);
```

Comment:
Just curious, is there a reason as to why this is used instead of the original retrieval method we used, for example: 

```typescript
const state = this.activatedRoute.snapshot.queryParamMap.get('state');
```

Reply:
Oh I didn't realised this `state` variable was being used to retrieve the URL... 😅  URLSearchParams has a bunch of methods that are convenient to extract the data needed thats why i used it

Reply:
Ahhh I see, for consistency sake, should we can just use the default `queryParamMap` to help us get the encoded URL vairables? What do you think!